src/wallet/rpcdump.cpp

```diff
@@ -854,211 +941,167 @@ static UniValue ProcessImport(CWallet * const pwallet, const UniValue& data, con
             if (!IsHex(output)) {
                 throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid scriptPubKey");
             }
-
             std::vector<unsigned char> vData(ParseHex(output));
             script = CScript(vData.begin(), vData.end());
             if (!ExtractDestination(script, dest) && !internal) {
                 throw JSONRPCError(RPC_INVALID_PARAMETER, "Internal must be set to true for nonstandard scriptPubKey imports.");
             }
         }
 
-        // Watchonly and private keys
-        if (watchOnly && keys.size()) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, "Watch-only addresses should not include private keys");
```

Comment:
Why was this check removed? Now it looks like if you provide a private key but set watchonly to true, it will still import that private key?

Reply:
Yes, how else would you import a private key for something you don't have all keys for? The old behavior was bogus.

Reply:
Ah, so now the only point of the watchonly parameter is to be explicit so we don't get a warning thrown

Reply:
I think the RPC help text for `watchonly` needs to be updated. It's currently `Stating whether matching outputs should be considered watched even when they're not spendable, only allowed if keys are empty`.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/14565#discussion_r239573986

> I think the RPC help text for watchonly needs to be updated.

I suggested one possible update in https://github.com/bitcoin/bitcoin/pull/14565#pullrequestreview-179073149:

> I think the documentation would be clearer if it just said that "watchonly" has no actual effect on the import, and only disables a warning if there are missing private keys.

Reply:
I'd rather not say it's just disabling a warning, as it may change to a failure later.